pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -319,6 +327,7 @@ func NewController(kubeClient kubelib.Client, options Options) *Controller {
 		})
 	})
 	registerHandlers(c.pods.informer, c.queue, "Pods", c.pods.onEvent, nil)
+	c.AppendWorkloadHandler(c.rebuildHostToNodeIndexOnPodEvent)
```

Comment:
We need to trigger  `c.xdsUpdater.SvcUpdate` in case any of the svcs external addresses changes from within rebuildHostToNodeIndexOnPodEvent?

I think, in the edge, we should stop triggereing cluster state push on EDSUpdates. @deva26

Reply:
Another way of handling pod events could have been from endpointcontroller, https://github.com/tetratelabs/istio/blob/master/pilot/pkg/serviceregistry/kube/controller/endpointcontroller.go#L90-L114

triggering EDSUpdate fromthere so may be....svcUpdate also from there..IDK, dont have a strong opinion between this way or from within endpoint controller. just thinking loud.

Reply:
> Another way of handling pod events could have been from endpointcontroller,

I see that I'm doing almost the same thing. So I'll move the logic here. Thanks for pointing out.

Reply:
> We need to trigger c.xdsUpdater.SvcUpdate in case any of the svcs external addresses changes from within rebuildHostToNodeIndexOnPodEvent?

Good catch. Added it. Yes. I think we need to trigger that as it affects the generated envoy cluster and (here) endpoint configs.